### PR TITLE
Simplify calico/node release process

### PR DESCRIPTION
**Pull request by bcreane**
## Description

Move the stream version (e.g. `v2.5`) from calico_node/Makefile into versions.yml. This simplifies the release process slightly since versions.yml becomes (mostly) the single source of truth for versions, and there's only one file to edit when managing calico/node versions (in this repo).
- testing: confirmed new version value is reflected in calico/node version; ran containerized fvs.

## Todos

- [x] Tests
- [x] Documentation

## Release Note

```release-note
None required
```

### CODE REVIEW

File: _data/versions.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+versions:
+  currentReleaseStream: v2.5
+  previousReleaseStream: v2.4
```
Comment (caseydavenport) [inline _data/versions.yml]: What is previousReleaseStream used for?

Comment (bcreane) [reply _data/versions.yml]: Nothing, currently. I've found it useful in some other experiments (swapping new release URLs for htmlproofer), but I'm happy to remove it for now if you prefer.

Comment (bcreane) [reply _data/versions.yml]: btw, I just https://github.com/projectcalico/calico/pull/932/files - I think this PR removes the need for that one.

Comment (caseydavenport) [reply _data/versions.yml]: Yeah, we shouldn't add stuff that isn't used. It's easy to add it in later if we find a real use for it.

I don't understand how this removes the need for the other PR - I think this just changes where the release stream is defined, yeah? There isn't actually any logic that changes here.

File: _data/versions.yml [hunk 2]
```diff
@@ -1,3 +1,7 @@
+versions:
+  currentReleaseStream: v2.5
```
Comment (caseydavenport) [inline _data/versions.yml]: Does this need to be a sub-field of versions? Could it be its own top-level field?

```yaml
currentReleaseStream: v2.5
versions:
  ...
```

File: _data/versions.yml (hunk 1 above)
Comment (bcreane) [reply _data/versions.yml]: Removed unused `previousReleaseStream` and moved `currentReleaseStream` to top level in versions.yml.
Re:  https://github.com/projectcalico/calico/pull/932/, you're right, this PR doesn't make a diff to that one.

File: calico_node/Makefile [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
-RELEASE_STREAM?=v2.5
+#
 
 GO_BUILD_VER?=v0.7
 CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)
 
 CALICO_NODE_DIR=$(dir $(realpath $(lastword $(MAKEFILE_LIST))))
 VERSIONS_FILE?=$(CALICO_NODE_DIR)/../_data/versions.yml
 
+# Read current and previous release stream version from _data/versions.yml
```
Comment (caseydavenport) [inline calico_node/Makefile]: Comment is out of date now

File: calico_node/Makefile [hunk 4]
```diff
@@ -1,13 +1,19 @@
 ###############################################################################
-# Versions:
-RELEASE_STREAM?=v2.5
+#
```
Comment (caseydavenport) [inline calico_node/Makefile]: I think this is a stray `#`?

**Comment by caseydavenport** [comment]
@bcreane one more comment and then could you squash commits?

File: calico_node/Makefile (hunk 4 above)
Comment (bcreane) [reply calico_node/Makefile]: yep! thanks for the Friday afternoon PR review :-)

File: RELEASING.md [hunk 5]
```diff
@@ -41,7 +41,8 @@ release is announced.
          values:
            version: vX.Y
     ```
-1. Update the `RELEASE_STREAM` variable at the top of calico_node/Makefile to the new major/minor version.
+1. Update `currentReleaseStream` and `previousReleaseStream` in `_data/versions.yml`.
```
Comment (bcreane) [inline RELEASING.md]: nuts. need one more update here.

**Comment by bcreane** [comment]
grr, semaphore. The build was indefinitely queued (20+ hours) so I deleted the branch on semaphore. After pushing a new commit to the PR on github, the branch isn't updating on semaphore (which is expected in hindsight). Unless @caseydavenport is willing to accept this PR w/o semaphore building it, then I'll likely close this PR and open another one.

File: _data/versions.yml [hunk 6]
```diff
@@ -1,3 +1,5 @@
+currentReleaseStream: v2.5  # vX.Y format: update during major release process
```
Comment (caseydavenport) [inline _data/versions.yml]: Very much a nitpick, but I'd rather not have an inline comment here, opting for the comment to come first instead.

Comment (bcreane) [reply _data/versions.yml]: Sure, happy to make the change. The comment was only my attempt at kicking semaphore out of its slumber. Looks like the right approach is to close/reopen the PR in github.

**Pull request merged**